obp/ope/classification_model.py

```diff
@@ -0,0 +1,660 @@
+# Copyright (c) Yuta Saito, Yusuke Narita, and ZOZO Technologies, Inc. All rights reserved.
+# Licensed under the Apache 2.0 License.
+
+"""Classification Model Class for Estimating Propensity Score and Importance Weight."""
+from dataclasses import dataclass
+from typing import Optional
+
+import numpy as np
+from sklearn.base import BaseEstimator
+from sklearn.base import clone
+from sklearn.model_selection import KFold
+from sklearn.utils import check_random_state
+from sklearn.utils import check_scalar
+from sklearn.calibration import CalibratedClassifierCV
+
+from ..utils import check_array, sample_action_fast, check_bandit_feedback_inputs
+
+
+@dataclass
+class ImportanceWeightEstimator(BaseEstimator):
+    """Machine learning model to distinguish between the behavior and evaluation policy (:math:`\\Pr[C = 1 | x, a]`),
+    where :math:`\\Pr[C=1|x,a]` is the probability that the data coming from the evaluation policy given action :math:`a` and :math:`x`.
+
+    Parameters
+    ------------
+    base_model: BaseEstimator
+        A machine learning model used to estimate the importance weights.
+
+    n_actions: int
+        Number of actions.
+
+    len_list: int, default=1
+        Length of a list of actions recommended in each impression.
+        When Open Bandit Dataset is used, 3 should be set.
+
+    action_context: array-like, shape (n_actions, dim_action_context), default=None
+        Context vector characterizing action (i.e., vector representation of each action).
+        If not given, one-hot encoding of the action variable is used as default.
+        If fitting_method is 'raw', action_context must be None.
```

Comment:
[IMO] We use only one-hot encoding style `action_context` in the current implementation, i.e., setting some special action contexts does not change the estimation results. Then, why not deleting this argument and use `self.action_context=self.action_context = np.eye(self.n_actions, dtype=int)`? I thought this extremely simplifies the current implementation while achieving the same procedure.